tests/actions/buildiso/buildiso_test.py

```diff
@@ -0,0 +1,164 @@
+import os
+
+import pytest
+
+from cobbler import enums
+from cobbler.actions import buildiso
+from cobbler.actions.buildiso.netboot import NetbootBuildiso
+from cobbler.actions.buildiso.standalone import StandaloneBuildiso
+from tests.conftest import does_not_raise
+
+
+@pytest.mark.parametrize(
+    "input_arch,result_binary_name,expected_exception",
+    [
+        (enums.Archs.X86_64, "grubx64.efi", does_not_raise()),
+        (enums.Archs.PPC, "grub.ppc64le", does_not_raise()),
+        (enums.Archs.PPC64, "grub.ppc64le", does_not_raise()),
+        (enums.Archs.PPC64EL, "grub.ppc64le", does_not_raise()),
+        (enums.Archs.PPC64LE, "grub.ppc64le", does_not_raise()),
+        (enums.Archs.AARCH64, "grubaa64.efi", does_not_raise()),
+        (enums.Archs.ARM, "bootarm.efi", does_not_raise()),
+        (enums.Archs.I386, "bootia32.efi", does_not_raise()),
+        (enums.Archs.IA64, "bootia64.efi", does_not_raise()),
+    ],
+)
+def test_calculate_grub_name(
+    input_arch,
+    result_binary_name,
+    expected_exception,
+    cobbler_api,
+    create_distro,
+):
+    # Arrange
+    test_builder = buildiso.BuildIso(cobbler_api)
+    test_distro = create_distro()
+    test_distro.arch = input_arch
+    cobbler_api.add_distro(test_distro)
+
+    # Act
+    with expected_exception:
+        result = test_builder.calculate_grub_name(test_distro)
+
+        # Assert
+        assert result == result_binary_name
+
+
+@pytest.mark.parametrize(
+    "input_kopts_dict,exepcted_output",
+    [
+        ({}, ""),
+        ({"test": 1}, " test=1"),
+        ({"test": None}, " test"),
+        ({"test": '"test"'}, ' test="test"'),
+        ({"test": "test test test"}, ' test="test test test"'),
+        ({"test": 'test "test" test'}, ' test="test "test" test"'),
+        ({"test": ['"test"']}, ' test="test"'),
+        ({"test": ['"test"', "test"]}, ' test="test" test=test'),
```

Comment:
Might be useful to also test a dict of dicts here, or is a dict as value of `input_kopts_dict` not possible?

Reply:
Indeed your suggestion is correct. This is not tested. However it is indeed not possible. These options are kernel options and the kernel options cannot have a nested dictionary. I will open an issue for this as the type hints need to be adjusted in a lot of places for this change. - See: https://github.com/cobbler/cobbler/issues/3212